src/Microsoft.DotNet.Darc/src/DarcLib/LocalGitClient.cs

```diff
@@ -467,6 +454,70 @@ private static void CleanRepoAndSubmodules(LibGit2Sharp.Repository repo, ILogger
             }
         }
 
+        private static void SafeCheckout(LibGit2Sharp.Repository repo, string commit, LibGit2Sharp.CheckoutOptions options, ILogger log)
+        {
+            try
+            {
+                log.LogDebug($"Trying safe checkout of {repo.Info.WorkingDirectory} at {commit}");
+                LibGit2Sharp.Commands.Checkout(repo, commit, options);
+            }
+            catch (Exception e) when (e is LibGit2Sharp.InvalidSpecificationException || e is LibGit2Sharp.NameConflictException || e is LibGit2Sharp.LibGit2SharpException)
+            {
+                log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files.");
+                SafeCheckoutByIndividualFiles(repo, commit, options, log);
+            }
+            catch
+            {
+                log.LogDebug($"Couldn't checkout {commit} as a commit after fetch.  Attempting to resolve as a treeish.");
+                string resolvedReference = ParseReference(repo, commit, log);
+                if (resolvedReference != null)
+                {
+                    log.LogDebug($"Resolved {commit} to {resolvedReference}, attempting to check out");
+                    try
+                    {
+                        log.LogDebug($"Trying checkout of {repo.Info.WorkingDirectory} at {resolvedReference}");
+                        LibGit2Sharp.Commands.Checkout(repo, resolvedReference, options);
+                    }
+                    catch (Exception e) when (e is LibGit2Sharp.InvalidSpecificationException || e is LibGit2Sharp.NameConflictException || e is LibGit2Sharp.LibGit2SharpException)
+                    {
+                        log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files.");
```

Comment:
>log.LogWarning($"Couldn't check out one or more files, possibly due to path length limitations ({e.ToString()}).  Attempting to checkout by individual files."); [](start = 24, length = 160)

Nit: split into shorter lines

Reply:
Changed as suggested.

---
In reply to: [285641411](https://github.com/dotnet/arcade-services/pull/422#discussion_r285641411) [](ancestors = 285641411)